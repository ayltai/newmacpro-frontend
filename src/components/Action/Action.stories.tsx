import { LaptopMac as MacIcon, PhoneIphone as IphoneIcon, } from '@mui/icons-material';
import { SpeedDialAction, } from '@mui/material';
import type { Meta, StoryObj, } from '@storybook/react';
import React from 'react';

import Action from './Action';

const meta : Meta<typeof Action> = {
    component : Action,
    tags      : [
        'autodocs',
    ],
    argTypes  : {
        mode     : {
            description : 'Specify the display mode to use. `default` is a button with text. `floating` is a floating action button.',
        },
        color    : {
            description : 'The color of the component. It supports both default and theme colors.',
        },
        icon     : {
            control     : false,
            description : 'An element placed before the content.',
        },
        badge    : {
            description : 'Specify the badge content to be displayed on the top-right corner of the component.',
        },
        children : {
            control     : false,
            description : 'The content of the component.',
        },
        onClick  : {
            description : 'An event handler called when the component is clicked.',
        },
    },
};

export default meta;

type Story = StoryObj<typeof Action>;

/**
 * An Action component with default props.
 */
export const Default : Story = {
    args : {
        mode     : 'default',
        color    : 'inherit',
        children : 'Default',
    },
};

/**
 * An Action component in `primary` color with an icon placed before the content.
 */
export const Icon : Story = {
    name : 'With icon',
    args : {
        ...Default.args,
        color    : 'primary',
        icon     : <MacIcon />,
        children : 'With icon',
    },
};

/**
 * An Action component showing as a floating action button in `primary` color with a badge placed on the top-right corner of the component.
 */
export const FloatingWithBadge : Story = {
    name       : 'Floating with badge',
    args       : {
        ...Icon.args,
        mode     : 'floating',
        icon     : <IphoneIcon />,
        badge    : 2,
        children : 'Checkout',
    },
    parameters : {
        docs : {
            story : {
                inline       : false,
                iframeHeight : 200,
            },
        },
    },
};

export const FloatingWithBadgeAndOptions : Story = {
    name       : 'Floating with badge and options',
    args       : {
        ...FloatingWithBadge.args,
        children : (
            <>
                <SpeedDialAction
                    key='Mac'
                    icon={<MacIcon />}
                    tooltipOpen
                    tooltipTitle='Mac' />
                <SpeedDialAction
                    key='iPhone'
                    icon={<IphoneIcon />}
                    tooltipOpen
                    tooltipTitle='iPhone' />
            </>
        ),
    },
    parameters : {
        docs : {
            story : {
                inline       : false,
                iframeHeight : 200,
            },
        },
    },
};
