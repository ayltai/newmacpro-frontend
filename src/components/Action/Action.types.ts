import { ReactNode, } from 'react';

export interface ActionProps {
    mode?            : 'default' | 'floating',
    color?           : 'inherit' | 'primary' | 'secondary',
    icon?            : ReactNode,
    badge?           : number,
    children?        : ReactNode,
    onClick?         : () => void,
    [ key : string ] : any,
}
