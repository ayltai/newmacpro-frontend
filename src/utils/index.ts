import * as Sentry from '@sentry/react';

export const handleError = (error : any) => {
    if (process.env.NODE_ENV === 'production') {
        Sentry.captureException(error);
    } else {
        console.error(error);
    }
};

export const capitalizeFirstLetter = (input : string) => input.charAt(0).toUpperCase() + input.slice(1);
