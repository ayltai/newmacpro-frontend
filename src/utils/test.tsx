import { ThemeProvider, } from '@mui/material';
import { configureStore, Store, } from '@reduxjs/toolkit';
import { act, render, } from '@testing-library/react';
import mediaQuery from 'css-mediaquery';
import { ConfirmProvider, } from 'material-ui-confirm';
import React, { ReactNode, } from 'react';
import { Provider, } from 'react-redux';

import { bundleApi, cartApi, } from '../apis';
import { POPULAR_APPS, } from '../constants';
import { cartReducer, prefReducer, store as reduxStore, } from '../states';
import { AppTheme, } from '../styles';
import { Cart, } from '../types';

export const mockApis = async (cart? : Cart) => {
    fetchMock.doMock(request => {
        if (request.url.endsWith('/api/bundles') && request.method === 'POST') return Promise.resolve({
            status : 201,
            body   : 'Created',
        });

        if (request.url.endsWith('/api/bundles/1') && request.method === 'GET') return Promise.resolve({
            status : 200,
            body   : JSON.stringify({
                id          : '1',
                displayName : 'Dummy',
                packages    : [
                    POPULAR_APPS[0],
                ],
            }),
        });

        if (request.url.endsWith('/api/bundles/1') && request.method === 'DELETE') return Promise.resolve({
            status : 200,
            body   : 'OK',
        });

        if (request.url.endsWith('/api/carts') && request.method === 'POST') return Promise.resolve({
            status : 201,
            body   : 'Created',
        });

        return Promise.resolve({
            status : 404,
            body   : 'Not found',
        });
    });

    await act(async () => {
        await reduxStore.dispatch(bundleApi.endpoints?.getBundle.initiate({
            bundleId : '1',
            idToken  : 'idToken',
        }));
        if (cart) await reduxStore.dispatch(cartApi.endpoints.addCart.initiate(cart));
    });
};

export const createMatchMedia = (width : number) => (query : string) => ({
    matches             : mediaQuery.match(query, {
        width,
    }),
    media               : '',
    onchange            : null,
    addListener         : () => {},
    removeListener      : () => {},
    addEventListener    : () => {},
    removeEventListener : () => {},
    dispatchEvent       : () => false,
});

export const createStore = (preloadedState? : Record<string, any>) => configureStore({
    preloadedState,
    reducer    : {
        cart                      : cartReducer,
        pref                      : prefReducer,
        [ bundleApi.reducerPath ] : bundleApi.reducer,
        [ cartApi.reducerPath   ] : cartApi.reducer,
    },
    middleware : getDefaultMiddleware => getDefaultMiddleware().concat(bundleApi.middleware, cartApi.middleware),
});

export const defaultStore = createStore();

export type RootState = ReturnType<typeof defaultStore.getState>;

const customRender = (ui : any, {
    preloadedState,
    store = createStore(preloadedState),
    ...rest
} : {
    preloadedState?   : Record<string, any>,
    store?            : Store<RootState>,
    [ rest : string ] : any,
} = {}) => render(ui, {
    wrapper : ({
        children,
    } : {
        children : ReactNode,
    }) => (
        <Provider store={store}>
            <ThemeProvider theme={AppTheme}>
                <ConfirmProvider>
                    {children}
                </ConfirmProvider>
            </ThemeProvider>
        </Provider>
    ),
    ...rest,
});

export * from '@testing-library/react';

export { customRender as render, };
