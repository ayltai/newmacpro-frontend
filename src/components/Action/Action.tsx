import { Badge, Button, Fab, IconButton, SpeedDial, useTheme, Zoom, } from '@mui/material';
import React, { Children, FC, useState, } from 'react';

import { ActionProps, } from './Action.types';

/**
 * Actions allow users to take actions, and make choices, with a single tap.
 * @param mode Specify the display mode to use. `default` is a button with text. `floating` is a floating action button.
 * @param color The color of the component. It supports both default and theme colors.
 * @param icon An element placed before the children.
 * @param badge Specify the badge content to be displayed on the top-right corner of the component.
 * @param children The content of the component.
 * @param onClick The event handler called when the component is clicked.
 * @param rest The other props passed down to the component.
 */
const Action : FC<ActionProps> = ({
    mode  = 'default',
    color = 'inherit',
    icon,
    badge,
    children,
    onClick,
    ...rest
}) => {
    const [ open, setOpen, ] = useState(false);

    const theme = useTheme();

    const transitionDuration = {
        enter : theme.transitions.duration.enteringScreen,
        exit  : theme.transitions.duration.leavingScreen,
    };

    const handleOpen = () => setOpen(true);

    const handleClose = () => setOpen(false);

    const defaultButton = !icon || children ? (
        <Button
            color={color}
            variant='contained'
            startIcon={icon}
            onClick={onClick}
            {...rest}>
            {children}
        </Button>
    ) : (
        <IconButton
            color={color}
            onClick={onClick}
            {...rest}>
            {icon}
        </IconButton>
    );

    const button = mode === 'floating' ? (
        <Zoom
            unmountOnExit
            style={{
                transitionDelay : `${badge ? 0 : transitionDuration.exit}ms`,
            }}
            in={!!badge}
            timeout={transitionDuration}>
            {Children.count(children) > 1 ? (
                <SpeedDial
                    sx={{
                        position : 'fixed',
                        right    : 32,
                        bottom   : 32,
                    }}
                    ariaLabel={rest.ariaLabel}
                    icon={icon}
                    open={open}
                    onOpen={handleOpen}
                    onClose={handleClose}>
                    {children}
                </SpeedDial>
            ) : (
                <Fab
                    sx={{
                        position : 'fixed',
                        right    : 32,
                        bottom   : 32,
                        zIndex   : 1100,
                    }}
                    color={color}
                    variant={children ? 'extended' : 'circular'}
                    onClick={onClick}
                    {...rest}>
                    {icon}
                    {children}
                </Fab>
            )}
        </Zoom>
    ) : defaultButton;

    return badge ? (
        <Badge
            sx={mode === 'floating' ? {
                position : 'fixed',
                right    : 32,
                bottom   : 92,
            } : undefined}
            color='secondary'
            badgeContent={badge}>
            {button}
        </Badge>
    ) : button;
};

export default Action;
