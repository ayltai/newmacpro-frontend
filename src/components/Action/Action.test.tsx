import { Laptop as LaptopIcon, } from '@mui/icons-material';
import React from 'react';

import { fireEvent, render, } from '../../utils/test';
import Action from './Action';

describe('<Action />', () => {
    it('renders default style correctly', () => {
        const handleClick = jest.fn();

        expect(render(<Action onClick={handleClick}>Dummy</Action>)).toMatchSnapshot();
    });

    it('renders floating style correctly', () => {
        const handleClick = jest.fn();

        expect(render(
            <Action
                mode='floating'
                onClick={handleClick}>
                Dummy
            </Action>
        )).toMatchSnapshot();
    });

    it('renders icon style correctly', () => {
        const handleClick = jest.fn();

        expect(render(
            <Action
                icon={<LaptopIcon />}
                onClick={handleClick}>
                Dummy
            </Action>
        )).toMatchSnapshot();
    });

    it('renders badge correctly', () => {
        const handleClick = jest.fn();

        expect(render(
            <Action
                badge={1}
                onClick={handleClick}>
                Dummy
            </Action>
        )).toMatchSnapshot();
    });

    it('handles click event correctly', () => {
        const handleClick = jest.fn();

        const { getByRole, } = render(
            <Action onClick={handleClick}>
                Dummy
            </Action>
        );

        fireEvent.click(getByRole('button'));

        expect(handleClick).toBeCalledTimes(1);
    });
});
