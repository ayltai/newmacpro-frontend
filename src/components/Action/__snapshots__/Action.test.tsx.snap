// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Action /> renders badge correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <span
        class="MuiBadge-root css-1c32n2y-MuiBadge-root"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedInherit MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorInherit MuiButton-root MuiButton-contained MuiButton-containedInherit MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorInherit css-j0lnzu-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Dummy
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <span
          class="MuiBadge-badge MuiBadge-standard MuiBadge-anchorOriginTopRight MuiBadge-anchorOriginTopRightRectangular MuiBadge-overlapRectangular MuiBadge-colorSecondary css-1sygzxv-MuiBadge-badge"
        >
          1
        </span>
      </span>
    </div>
  </body>,
  "container": <div>
    <span
      class="MuiBadge-root css-1c32n2y-MuiBadge-root"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedInherit MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorInherit MuiButton-root MuiButton-contained MuiButton-containedInherit MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorInherit css-j0lnzu-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Dummy
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <span
        class="MuiBadge-badge MuiBadge-standard MuiBadge-anchorOriginTopRight MuiBadge-anchorOriginTopRightRectangular MuiBadge-overlapRectangular MuiBadge-colorSecondary css-1sygzxv-MuiBadge-badge"
      >
        1
      </span>
    </span>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<Action /> renders default style correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedInherit MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorInherit MuiButton-root MuiButton-contained MuiButton-containedInherit MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorInherit css-j0lnzu-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Dummy
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedInherit MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorInherit MuiButton-root MuiButton-contained MuiButton-containedInherit MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorInherit css-j0lnzu-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      Dummy
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<Action /> renders floating style correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div />
  </body>,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<Action /> renders icon style correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedInherit MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorInherit MuiButton-root MuiButton-contained MuiButton-containedInherit MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorInherit css-j0lnzu-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-startIcon MuiButton-iconSizeMedium css-gcc2o7-MuiButton-startIcon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="LaptopIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M20 18c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2H4c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2H0v2h24v-2h-4zM4 6h16v10H4V6z"
            />
          </svg>
        </span>
        Dummy
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedInherit MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorInherit MuiButton-root MuiButton-contained MuiButton-containedInherit MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorInherit css-j0lnzu-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-startIcon MuiButton-iconSizeMedium css-gcc2o7-MuiButton-startIcon"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="LaptopIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 18c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2H4c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2H0v2h24v-2h-4zM4 6h16v10H4V6z"
          />
        </svg>
      </span>
      Dummy
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
